diff --git a/meson.build b/meson.build
index 6032e70..6fb0aa2 100644
--- a/meson.build
+++ b/meson.build
@@ -21,7 +21,8 @@ else
   )
 endif
 
-fmt_dep = dependency('fmt')
+fmt_dep = dependency('fmt', static: true)
+add_project_arguments('-DFMT_DEPRECATED_HEAVY_CORE', language: ['c', 'cpp'])
 
 if host_machine.system() != 'windows'
   thread_dep = dependency('threads')
@@ -89,7 +90,7 @@ solaris_dep = dependency(
 
 if host_machine.system() == 'windows'
   add_project_arguments(
-    '-DNOMINMAX',
+    ['-DNOMINMAX', '-D_DLL', '-D_MT'],
     language: 'cpp',
   )
   if cpp.get_argument_syntax() == 'msvc'
@@ -102,6 +103,11 @@ if host_machine.system() == 'windows'
       language: 'cpp',
     )
   endif
+  if get_option('buildtype') == 'debug'
+    add_project_arguments(['-D_ITERATOR_DEBUG_LEVEL=2', '-fms-runtime-lib=dll_dbg'], language: 'cpp')
+  else
+    add_project_arguments('-fms-runtime-lib=dll', language: 'cpp')
+  endif
 endif
 
 add_project_arguments(
