diff --git a/simd/meson.build b/simd/meson.build
index a870493..7b07d10 100644
--- a/simd/meson.build
+++ b/simd/meson.build
@@ -148,7 +148,7 @@ if host_cpu in ['x86', 'x86_64']
       dir / 'jsimd.c',
       simd_src,
       include_directories: [incdir, 'nasm', dir],
-      pic: get_option('default_library') != 'static',
+      pic: true,
     )
   endif
 elif host_cpu in ['arm', 'aarch64']
@@ -352,7 +352,7 @@ elif host_cpu in ['arm', 'aarch64']
     simd = static_library(
       'simd',
       simd_src_arm + files(dir / 'jsimd.c'),
-      pic: get_option('default_library') != 'static',
+      pic: true,
       include_directories: [incdir, 'arm', dir],
       c_args: neon_flags,
     )
@@ -377,7 +377,7 @@ elif host_cpu in ['ppc', 'ppc64']
       dir / 'jsimd.c',
       simd_src_ppc,
       include_directories: [incdir, dir],
-      pic: get_option('default_library') != 'static',
+      pic: true,
       c_args: '-maltivec',
     )
   endif
diff --git a/src/meson.build b/src/meson.build
index ad228f8..968b392 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -138,6 +138,7 @@ jpeg = library(
   soversion: so_version,
   vs_module_defs: vs_defs,
   install: true,
+  pic: true
 )
 
 pkg.generate(
@@ -176,6 +177,7 @@ if get_option('turbojpeg').allowed()
     install: true,
     link_whole: simd,
     soversion: '0.4.0',
+    pic: true
   )
 
   pkg.generate(
