diff --git a/tools/slang-fiddle/slang-fiddle-lua.cpp b/tools/slang-fiddle/slang-fiddle-lua.cpp
index 882289ca4..ddcff53d5 100644
--- a/tools/slang-fiddle/slang-fiddle-lua.cpp
+++ b/tools/slang-fiddle/slang-fiddle-lua.cpp
@@ -7,4 +7,4 @@
 #define LUA_USE_POSIX
 #endif
 
-#include "lua/onelua.c"
+#include "onelua.c"
diff --git a/tools/slang-fiddle/slang-fiddle-scrape.h b/tools/slang-fiddle/slang-fiddle-scrape.h
index 098c11aad..557e0e7a6 100644
--- a/tools/slang-fiddle/slang-fiddle-scrape.h
+++ b/tools/slang-fiddle/slang-fiddle-scrape.h
@@ -2,7 +2,7 @@
 #pragma once
 
 #include "compiler-core/slang-lexer.h"
-#include "lua/lauxlib.h"
+#include "lauxlib.h"
 #include "slang-fiddle-diagnostics.h"
 
 namespace fiddle
diff --git a/tools/slang-fiddle/slang-fiddle-script.cpp b/tools/slang-fiddle/slang-fiddle-script.cpp
index eb55e383d..82293ff77 100644
--- a/tools/slang-fiddle/slang-fiddle-script.cpp
+++ b/tools/slang-fiddle/slang-fiddle-script.cpp
@@ -2,9 +2,9 @@
 #include "slang-fiddle-script.h"
 
 #include "compiler-core/slang-diagnostic-sink.h"
-#include "lua/lapi.h"
-#include "lua/lauxlib.h"
-#include "lua/lualib.h"
+#include "lapi.h"
+#include "lauxlib.h"
+#include "lualib.h"
 
 #include <cstdio>
 
diff --git a/tools/slang-fiddle/slang-fiddle-script.h b/tools/slang-fiddle/slang-fiddle-script.h
index 94bda453b..4432fe0e5 100644
--- a/tools/slang-fiddle/slang-fiddle-script.h
+++ b/tools/slang-fiddle/slang-fiddle-script.h
@@ -4,8 +4,8 @@
 #include "compiler-core/slang-source-loc.h"
 #include "core/slang-list.h"
 #include "core/slang-string.h"
-#include "lua/lapi.h"
-#include "lua/lauxlib.h"
+#include "lapi.h"
+#include "lauxlib.h"
 #include "slang-fiddle-diagnostics.h"
 #include "slang-fiddle-scrape.h"
 
