diff --git a/CMakeLists.txt b/CMakeLists.txt
index 265266d..8bc588d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -486,6 +486,101 @@ endif()
 
 fetch_or_build_slang_llvm()
 
+#
+# generate capability files
+#
+glob_append(SLANG_CAPABILITY_SOURCE "source/slang/*.capdef")
+
+set(SLANG_CAPABILITY_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/capability")
+
+# Generate the output file list
+set(SLANG_CAPABILITY_GENERATED_HEADERS
+    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-generated-capability-defs.h"
+    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-generated-capability-defs-impl.h"
+)
+set(SLANG_CAPABILITY_GENERATED_SOURCE
+    ${SLANG_CAPABILITY_GENERATED_HEADERS}
+    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-lookup-capability-defs.cpp"
+)
+add_custom_command(
+    OUTPUT ${SLANG_CAPABILITY_GENERATED_SOURCE}
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${SLANG_CAPABILITY_OUTPUT_DIR}
+    COMMAND
+        slang-capability-generator ${SLANG_CAPABILITY_SOURCE} --target-directory
+        ${SLANG_CAPABILITY_OUTPUT_DIR} --doc
+        "${slang_SOURCE_DIR}/docs/user-guide/a3-02-reference-capability-atoms.md"
+    DEPENDS ${SLANG_CAPABILITY_SOURCE} slang-capability-generator
+    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+    VERBATIM
+)
+
+#
+# generated lookup tables
+#
+
+get_target_property(
+    SLANG_SPIRV_HEADERS_INCLUDE_DIR
+    SPIRV-Headers::SPIRV-Headers
+    INTERFACE_INCLUDE_DIRECTORIES
+)
+
+set(SLANG_LOOKUP_GENERATOR_INPUT_JSON
+    "${SLANG_SPIRV_HEADERS_INCLUDE_DIR}/spirv/unified1/extinst.glsl.std.450.grammar.json"
+)
+set(SLANG_LOOKUP_GENERATOR_OUTPUT_DIR
+    "${CMAKE_CURRENT_BINARY_DIR}/slang-lookup-tables/"
+)
+set(SLANG_LOOKUP_GENERATED_SOURCE
+    "${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}/slang-lookup-GLSLstd450.cpp"
+)
+set(SLANG_RECORD_REPLAY_SYSTEM
+    "${slang_SOURCE_DIR}/source/slang-record-replay/record"
+    "${slang_SOURCE_DIR}/source/slang-record-replay/util"
+)
+
+add_custom_command(
+    OUTPUT ${SLANG_LOOKUP_GENERATED_SOURCE}
+    COMMAND
+        ${CMAKE_COMMAND} -E make_directory ${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}
+    COMMAND
+        slang-lookup-generator ${SLANG_LOOKUP_GENERATOR_INPUT_JSON}
+        ${SLANG_LOOKUP_GENERATED_SOURCE} "GLSLstd450" "GLSLstd450"
+        "spirv/unified1/GLSL.std.450.h"
+    DEPENDS ${SLANG_LOOKUP_GENERATOR_INPUT_JSON} slang-lookup-generator
+    VERBATIM
+)
+
+set(SLANG_SPIRV_CORE_SOURCE_JSON
+    "${SLANG_SPIRV_HEADERS_INCLUDE_DIR}/spirv/unified1/spirv.core.grammar.json"
+)
+set(SLANG_SPIRV_CORE_GRAMMAR_SOURCE
+    "${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}/slang-spirv-core-grammar-embed.cpp"
+)
+add_custom_command(
+    OUTPUT ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
+    COMMAND
+        ${CMAKE_COMMAND} -E make_directory ${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}
+    COMMAND
+        slang-spirv-embed-generator ${SLANG_SPIRV_CORE_SOURCE_JSON}
+        ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
+    DEPENDS ${SLANG_SPIRV_CORE_SOURCE_JSON} slang-spirv-embed-generator
+    VERBATIM
+)
+
+slang_add_target(
+    slang-lookup-tables
+    STATIC
+    EXPORT_MACRO_PREFIX SLANG
+    USE_EXTRA_WARNINGS
+    EXPLICIT_SOURCE
+        ${SLANG_LOOKUP_GENERATED_SOURCE}
+        ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
+    EXPORT_TYPE_AS ${SLANG_LIB_TYPE}
+    LINK_WITH_PRIVATE core SPIRV-Headers::SPIRV-Headers
+    EXCLUDE_FROM_ALL
+    FOLDER generated
+)
+
 #
 # Our targets
 #
diff --git a/prelude/CMakeLists.txt b/prelude/CMakeLists.txt
index 913809f..7347204 100644
--- a/prelude/CMakeLists.txt
+++ b/prelude/CMakeLists.txt
@@ -22,7 +22,7 @@ endforeach()
 
 slang_add_target(
     .
-    OBJECT
+    STATIC
     EXPORT_MACRO_PREFIX SLANG
     EXPLICIT_SOURCE ${prelude_source}
     EXCLUDE_FROM_ALL
diff --git a/source/core/slang-io.cpp b/source/core/slang-io.cpp
index 36c8c56..fb01a76 100644
--- a/source/core/slang-io.cpp
+++ b/source/core/slang-io.cpp
@@ -752,7 +752,7 @@ String Path::getRelativePath(String base, String path)
     auto result = std::filesystem::relative(p2, p1, ec);
     if (ec)
         return path;
-    return String(reinterpret_cast<const char*>(result.generic_u8string().c_str()));
+    return String(reinterpret_cast<const char*>(result.generic_string().c_str()));
 }
 
 SlangResult Path::remove(const String& path)
diff --git a/source/slang-core-module/CMakeLists.txt b/source/slang-core-module/CMakeLists.txt
index 461908c..ab91bab 100644
--- a/source/slang-core-module/CMakeLists.txt
+++ b/source/slang-core-module/CMakeLists.txt
@@ -54,7 +54,7 @@ add_custom_command(
 
 set(core_module_source_common_args
     .
-    OBJECT
+    STATIC
     EXCLUDE_FROM_ALL
     USE_EXTRA_WARNINGS
     LINK_WITH_PRIVATE
@@ -69,6 +69,7 @@ set(core_module_source_common_args
     generated
     EXPLICIT_SOURCE
     ./slang-embedded-core-module-source.cpp
+    ${SLANG_CAPABILITY_GENERATED_SOURCE}
     EXPORT_MACRO_PREFIX
     SLANG
     EXPORT_TYPE_AS
@@ -132,7 +133,7 @@ set_target_properties(generate_core_module_headers PROPERTIES FOLDER generated)
 
 set(core_module_common_args
     .
-    OBJECT
+    STATIC
     EXCLUDE_FROM_ALL
     EXPORT_MACRO_PREFIX
     SLANG
diff --git a/source/slang/CMakeLists.txt b/source/slang/CMakeLists.txt
index e5943bd..52c329c 100644
--- a/source/slang/CMakeLists.txt
+++ b/source/slang/CMakeLists.txt
@@ -49,33 +49,6 @@ target_include_directories(
     INTERFACE ${SLANG_FIDDLE_OUTPUT_DIR}
 )
 
-#
-# generate capability files
-#
-glob_append(SLANG_CAPABILITY_SOURCE "*.capdef")
-
-set(SLANG_CAPABILITY_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/capability")
-
-# Generate the output file list
-set(SLANG_CAPABILITY_GENERATED_HEADERS
-    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-generated-capability-defs.h"
-    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-generated-capability-defs-impl.h"
-)
-set(SLANG_CAPABILITY_GENERATED_SOURCE
-    ${SLANG_CAPABILITY_GENERATED_HEADERS}
-    "${SLANG_CAPABILITY_OUTPUT_DIR}/slang-lookup-capability-defs.cpp"
-)
-add_custom_command(
-    OUTPUT ${SLANG_CAPABILITY_GENERATED_SOURCE}
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${SLANG_CAPABILITY_OUTPUT_DIR}
-    COMMAND
-        slang-capability-generator ${SLANG_CAPABILITY_SOURCE} --target-directory
-        ${SLANG_CAPABILITY_OUTPUT_DIR} --doc
-        "${slang_SOURCE_DIR}/docs/user-guide/a3-02-reference-capability-atoms.md"
-    DEPENDS ${SLANG_CAPABILITY_SOURCE} slang-capability-generator
-    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-    VERBATIM
-)
 slang_add_target(
     slang-capability-defs
     OBJECT
@@ -92,7 +65,7 @@ slang_add_target(
 )
 slang_add_target(
     slang-capability-lookup
-    OBJECT
+    STATIC
     EXPORT_MACRO_PREFIX SLANG
     USE_EXTRA_WARNINGS
     EXPLICIT_SOURCE ${SLANG_CAPABILITY_GENERATED_SOURCE}
@@ -102,73 +75,6 @@ slang_add_target(
     FOLDER generated
 )
 
-#
-# generated lookup tables
-#
-
-get_target_property(
-    SLANG_SPIRV_HEADERS_INCLUDE_DIR
-    SPIRV-Headers::SPIRV-Headers
-    INTERFACE_INCLUDE_DIRECTORIES
-)
-
-set(SLANG_LOOKUP_GENERATOR_INPUT_JSON
-    "${SLANG_SPIRV_HEADERS_INCLUDE_DIR}/spirv/unified1/extinst.glsl.std.450.grammar.json"
-)
-set(SLANG_LOOKUP_GENERATOR_OUTPUT_DIR
-    "${CMAKE_CURRENT_BINARY_DIR}/slang-lookup-tables/"
-)
-set(SLANG_LOOKUP_GENERATED_SOURCE
-    "${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}/slang-lookup-GLSLstd450.cpp"
-)
-set(SLANG_RECORD_REPLAY_SYSTEM
-    "${slang_SOURCE_DIR}/source/slang-record-replay/record"
-    "${slang_SOURCE_DIR}/source/slang-record-replay/util"
-)
-
-add_custom_command(
-    OUTPUT ${SLANG_LOOKUP_GENERATED_SOURCE}
-    COMMAND
-        ${CMAKE_COMMAND} -E make_directory ${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}
-    COMMAND
-        slang-lookup-generator ${SLANG_LOOKUP_GENERATOR_INPUT_JSON}
-        ${SLANG_LOOKUP_GENERATED_SOURCE} "GLSLstd450" "GLSLstd450"
-        "spirv/unified1/GLSL.std.450.h"
-    DEPENDS ${SLANG_LOOKUP_GENERATOR_INPUT_JSON} slang-lookup-generator
-    VERBATIM
-)
-
-set(SLANG_SPIRV_CORE_SOURCE_JSON
-    "${SLANG_SPIRV_HEADERS_INCLUDE_DIR}/spirv/unified1/spirv.core.grammar.json"
-)
-set(SLANG_SPIRV_CORE_GRAMMAR_SOURCE
-    "${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}/slang-spirv-core-grammar-embed.cpp"
-)
-add_custom_command(
-    OUTPUT ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
-    COMMAND
-        ${CMAKE_COMMAND} -E make_directory ${SLANG_LOOKUP_GENERATOR_OUTPUT_DIR}
-    COMMAND
-        slang-spirv-embed-generator ${SLANG_SPIRV_CORE_SOURCE_JSON}
-        ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
-    DEPENDS ${SLANG_SPIRV_CORE_SOURCE_JSON} slang-spirv-embed-generator
-    VERBATIM
-)
-
-slang_add_target(
-    slang-lookup-tables
-    OBJECT
-    EXPORT_MACRO_PREFIX SLANG
-    USE_EXTRA_WARNINGS
-    EXPLICIT_SOURCE
-        ${SLANG_LOOKUP_GENERATED_SOURCE}
-        ${SLANG_SPIRV_CORE_GRAMMAR_SOURCE}
-    EXPORT_TYPE_AS ${SLANG_LIB_TYPE}
-    LINK_WITH_PRIVATE core SPIRV-Headers::SPIRV-Headers
-    EXCLUDE_FROM_ALL
-    FOLDER generated
-)
-
 #
 # Generate the version header
 #
@@ -269,19 +175,22 @@ else()
     # the shared objects as one target so we don't build them twice.
     slang_add_target(
         .
-        OBJECT
+        STATIC
         ${slang_build_args}
         ${slang_link_args}
         TARGET_NAME slang-common-objects
         EXCLUDE_FROM_ALL
     )
+    target_sources(slang-common-objects PUBLIC ${SLANG_REFLECT_GENERATED_HEADERS} ${SLANG_CAPABILITY_GENERATED_HEADERS})
     slang_add_target(
         .
         ${SLANG_LIB_TYPE}
         EXPORT_MACRO_PREFIX SLANG
         ${slang_link_args}
         ${slang_interface_args}
-        NO_SOURCE
+        EXPLICIT_SOURCE
+        dummy.cpp
+        EXTRA_COMPILE_DEFINITIONS_PRIVATE DUMMY_A
         TARGET_NAME slang-without-embedded-core-module
         EXCLUDE_FROM_ALL
         LINK_WITH_PRIVATE
@@ -301,6 +210,8 @@ else()
         ${slang_interface_args}
         ${slang_public_lib_args}
         LINK_WITH_PRIVATE slang-common-objects
-        NO_SOURCE
+        EXPLICIT_SOURCE
+        dummy.cpp
+        EXTRA_COMPILE_DEFINITIONS_PRIVATE DUMMY_B
     )
 endif()
diff --git a/source/slang/dummy.cpp b/source/slang/dummy.cpp
new file mode 100644
index 0000000..78f9ee5
--- /dev/null
+++ b/source/slang/dummy.cpp
@@ -0,0 +1,6 @@
+#ifdef DUMMY_A
+void dum_a() {}
+#endif
+#ifdef DUMMY_B
+void dum_b() {}
+#endif
\ No newline at end of file
