diff --git a/libtiff/meson.build b/libtiff/meson.build
index aa1d96d..697722d 100644
--- a/libtiff/meson.build
+++ b/libtiff/meson.build
@@ -76,7 +76,7 @@ sources = files(
 )
 
 if host_machine.system() == 'windows'
-  sources += import('windows').compile_resources('tif_win32_versioninfo.rc')
+  sources += import('windows').compile_resources('tif_win32_versioninfo.rc', depend_files: tiffvers_h)
 endif
 
 tiff4_lib = library(
@@ -97,4 +97,27 @@ tiff4_lib = library(
   version: library_version,
 )
 
+defs = []
+if host_machine.system() == 'windows'
+  defs += '-D_DLL'
+  defs += '-D_MT'
+  if get_option('buildtype') == 'debug'
+    defs += '-D_DEBUG'
+    defs += '-D_ITERATOR_DEBUG_LEVEL=2'
+    defs += '-fms-runtime-lib=dll_dbg'
+  else
+    defs += '-fms-runtime-lib=dll'
+  endif
+endif
+
+tiff4xx_lib = library(
+  'tiffxx',
+  'tif_stream.cxx',
+  install: true,
+  dependencies: [m_dep, jbig_dep, jpeg_dep, lerc_dep, lzma_dep, webp_dep, zlib_dep, zstd_dep],
+  link_whole: tiff4_lib,
+  cpp_args: defs
+)
+
+install_headers('tiffio.hxx')
 install_headers('tiff.h', 'tiffio.h', tiffconf_h, tiffvers_h)
diff --git a/meson.build b/meson.build
index f457788..541591f 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,7 @@
 project(
   'tiff',
   'c',
+  'cpp',
   version: files('VERSION'),
   license: 'libtiff AND BSD-4.3TAHOE',
   meson_version: '>=0.57.0',
@@ -176,6 +177,11 @@ libtiff4_dep = declare_dependency(
 
 meson.override_dependency('libtiff-4', libtiff4_dep)
 
+libtiff4xx_dep = declare_dependency(
+  include_directories: 'libtiff',
+  link_with: tiff4xx_lib
+)
+
 import('pkgconfig').generate(
   tiff4_lib,
   filebase: 'libtiff-4',
